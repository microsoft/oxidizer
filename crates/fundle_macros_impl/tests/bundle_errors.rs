// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use syn::{ItemStruct, parse_quote};

mod util;


#[test]
fn forward_empty() {
    let item: ItemStruct = parse_quote! {
        #[bundle]
        struct Foo {
            #[forward()]
            x: Bar
        }
    };

    insta::assert_snapshot!(expand_fundle_bundle!(item));
}

#[test]
fn forward_invalid_syntax() {
    let item: ItemStruct = parse_quote! {
        #[bundle]
        struct Foo {
            #[forward(123)]
            x: Bar
        }
    };

    insta::assert_snapshot!(expand_fundle_bundle!(item));
}
