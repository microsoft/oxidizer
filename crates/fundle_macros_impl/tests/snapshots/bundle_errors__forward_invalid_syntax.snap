---
source: crates/fundle_macros_impl/tests/bundle_errors.rs
expression: expand_fundle_bundle!(item)
---
::core::compile_error! {
    "fundle::bundle #[forward(...)] attribute must contain valid type paths"
}
